internal/app/machined/pkg/runtime/v1alpha1/v1alpha1_sequencer.go

```diff
@@ -287,6 +287,20 @@ func (*Sequencer) Boot(r runtime.Runtime) []runtime.Phase {
 	return phases
 }
 
+func parseWipeParamString(wipe *string) []runtime.Phase {
+	if wipe == nil || *wipe == "" {
+		return nil
+	}
+
+	phases := PhaseList{}
+
+	if *wipe == "system" {
+		return phases.Append("wipeSystemDisk", ResetSystemDisk).Append("reboot", Reboot)
```

Comment:
I think instead of doing all these surgery here, we could instead add a phase which runs only if the `wipe` parameter is set, and does nothing otherwise. we could call `Reboot` from the task itself as well? Reboot returns an error and aborts the sequencer iirc

Reply:
Fixed. `ResetSystemDiskPartitions` becomes kinds complex, returning task function depending on `talos.experimental.wipe`, but maybe that's okay?